spec/lib/email/receiver_spec.rb

```diff
@@ -756,6 +756,16 @@ def create_post_reply_key(value)
       MD
     end
 
+    it "tries not to repeat duplicate attachments" do
+      SiteSetting.authorized_extensions = "jpg"
+
+      expect { process(:logo_1) }.to change { UploadReference.count }.by(1)
+      expect(topic.posts.last.raw).to match %r{upload://}
+
+      expect { process(:logo_2) }.not_to change { UploadReference.count }
+      expect(topic.posts.last.raw).not_to match %r{upload://}
```

Comment:
Safety check to ensure the attachment is **not** present in the raw of that 2nd post.